### DISCUSSION TITLE
Enum Validation Error Not Caught On Testing

### DISCUSSION THREAD

**Discussion by bzarzuela** [livewire/livewire#8447] [7. Bugs (archived)]
### Livewire version

v3.4.12

### Laravel version

v11.7.0

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a form with a property that is cast as an Enum in Eloquent with an `'In` validator .

```
    #[Validate(['nullable', 'in:High,Low'])]
    public $priority = '';
```

2. Unit testing editing works

```
$component = Livewire::actingAs($user)
        ->test(Edit::class, ['ticket' => $ticket])
        ->set('form.priority', Priority::High)
        ->call('save')
        ->assertHasNoErrors();
```

3. Trying to test the form manually yields an `Object of class App\Tickets\Enums\Priority could not be converted to string` error

### Repository or Wirebox URL

https://wirebox.app/b/x5lww

### Additional information

I'm sure it's because I should be using an Enum validator on the form but the fact that it passed the unit test means that code like this may pass automated build pipelines.

I hope there's a way for Livewire to catch it or maybe mention in the docs to be careful about cases like this. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PerryvanderMeer** [comment]
I would suggest using Laravel's [`Enum` validation rule](https://laravel.com/docs/11.x/validation#rule-enum), because the `in` rule tries to type-hint variables into strings:

https://github.com/laravel/framework/blob/bf7046f37a03ac006f60a811148f91393f305283/src/Illuminate/Validation/Concerns/ValidatesAttributes.php#L1373C47-L1373C55
